developer_manual/digging_deeper/dashboard.rst

```diff
@@ -0,0 +1,150 @@
+=========
+Dashboard
+=========
+
+The dashboard app aims to provide the user with a general overview of their 
+Nextcloud and show information that is currently important.
+
+App developers can integrate into the dashboard app and provide their own panels.
+
+
+Register a dashboard panels
+===========================
+
+A dashboard panel is represented by a class implementing the `OCP\\Dashboard\\IPanel` 
+interface. This class will be instantiated whenever the dashboard is loaded, so 
+scripts and initial state that your panel needs can be added in the constructor.
```

Comment:
I guess this should be raised in the PR rather than here, but I personally don't like seeing this kind of *active code* in a constructor. In my understanding, constructors should only do the things necessary to initialize the newly constructed object.

This pattern makes it also a bit harder/weirder to unit test. As soon as you create an object, a side effect happens.

An alternative would be do have like a dedicated method ``load`` or something that is invoked when apps should run these tasks (initial state, scripts, ...). Would that make sense?

Reply:
Adjusted the server PR and docs here